casr/src/bin/casr-cluster.rs

```diff
@@ -70,21 +72,29 @@ fn make_clusters(inpath: &Path, outpath: Option<&Path>, jobs: usize) -> Result<u
 
     // Stacktraces from casreps
     let traces: RwLock<Vec<Stacktrace>> = RwLock::new(Vec::new());
+    // Crashlines from casreps
+    let crashlines: RwLock<Vec<String>> = RwLock::new(Vec::new());
     // Casreps with stacktraces, that we can parse
     let filtered_casreps: RwLock<Vec<PathBuf>> = RwLock::new(Vec::new());
     // Casreps with stacktraces, that we cannot parse
     let mut badreports: RwLock<Vec<PathBuf>> = RwLock::new(Vec::new());
     custom_pool.install(|| {
         (0..len).into_par_iter().for_each(|i| {
-            if let Ok(trace) = stacktrace(casreps[i].as_path()) {
-                traces.write().unwrap().push(trace);
-                filtered_casreps.write().unwrap().push(casreps[i].clone());
+            if let Ok(report) = util::report_from_file(casreps[i].as_path()) {
+                if let Ok(trace) = report.filtered_stacktrace() {
```

Comment:
Add to badreports when filtered_stacktrace fails

Reply:
Done